### DISCUSSION TITLE
Redistributable-less version of otd?

### DISCUSSION THREAD

**Discussion by WalkTheEarth** [OpenTabletDriver/OpenTabletDriver#3519] [General]
### Description

Hi all, i would like to know if it is possible to make a redist-less version of opentabletdriver? (or atleast a way to run it without admin, my school laptop (windows) dont got admin perms) 

### Acknowledgements

- [X] I have searched the existing issues and this new issue is not a duplicate of any.
- [X] I have written a concise and meaningful title.
- [ ] I am on the latest version of OpenTabletDriver.

**Comment by WalkTheEarth** [comment]
what the hell is `needs-triage`

**Comment by jamesbt365** [comment]
The only thing that needs admin for OpenTabletDriver is installing dotnet (and cleaning up other drivers). You can compile your own version of OpenTabletDriver that is self contained by modifying the [build scripts](https://github.com/OpenTabletDriver/OpenTabletDriver/blob/0.6.x/eng/windows/package.ps1#L9).

You can either fork this repo, change it on the 0.6.x branch and run the release CI or simply do the change on the line i linked before and then run the build script in the base of the repo.

**Reply by Mrcubix** [reply]
In addition, dotnet can be installed in any non-admin directory, and provided you set `DOTNET_ROOT` to the folder in question, it will work the same.

**Comment by gonX** [comment]
Related: #1442

For what it's worth, our OSX release already uses self-contained publishing, and the file sizes differences have been significantly reduced with .NET 6.0.
